walletlibrary/src/main/java/com/microsoft/walletlibrary/util/http/httpagent/IResponse.kt

```diff
@@ -1,7 +1,68 @@
 package com.microsoft.walletlibrary.util.http.httpagent
 
+import com.google.common.net.HttpHeaders
+import com.microsoft.walletlibrary.util.NetworkingException
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import com.microsoft.walletlibrary.did.sdk.util.Constants
+
 class IResponse(
     val status: Int,
     val headers: Map<String, String>,
     val body: ByteArray
-)
\ No newline at end of file
+) {
+    fun toNetworkingException(): NetworkingException {
+        var code = status.toString()
+        var innerError: String? = null
+        var message = ""
+        var correlationId: String? = headers["ms-cv"]
+        var bodyAsString: String? = null
+        val retryable: Boolean = headers.containsKey(HttpHeaders.RETRY_AFTER)
+        try {
+            bodyAsString = body.decodeToString()
+            val json = Json.decodeFromString<JsonObject>(bodyAsString)
+            if (correlationId == null) {
+                correlationId = (json[Constants.CORRELATION_VECTOR_HEADER] as? JsonPrimitive)?.content
+            }
+            // Attempt to parse assuming https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#handling-errors
```

Comment:
There was a parser to parse inner error in VC SDK. Could we reuse it?

Reply:
Thats actually where I pulled this code from. The error parser there only pulls one field from each innerError object though so I figured it'd be more efficient to handle both during traversal.